Removed product_start.ycp file is obsolete according to Jiří Suchomel.

TODO: Upstream this.

diff --git a/agents/Makefile.am b/agents/Makefile.am
index 112eb8f..e69de29 100644
--- a/agents/Makefile.am
+++ b/agents/Makefile.am
@@ -1,7 +0,0 @@
-# Makefile.am for firstboot/agents
-
-agent_SCRIPTS =
-
-scrconf_DATA = cfg_firstboot.scr
-
-EXTRA_DIST = $(agent_SCRIPTS) $(scrconf_DATA)
diff --git a/desktop/Makefile.am b/desktop/Makefile.am
index 1f8f3a5..832dcf6 100644
--- a/desktop/Makefile.am
+++ b/desktop/Makefile.am
@@ -1,15 +1,9 @@
 # Makefile.am for firstboot/src/data
 
 
-fillupdir = /var/adm/fillup-templates/
-fillup_DATA = sysconfig.firstboot
-
 modconfigdir = $(prefix)/share/autoinstall/modules
 modconfig_DATA = firstboot.desktop
 
-schemafilesdir =  $(schemadir)/autoyast/rnc
-schemafiles_DATA = $(wildcard *.rnc)
-
-EXTRA_DIST = $(fillup_DATA) $(modconfig_DATA) $(schemafiles_DATA)
+EXTRA_DIST = $(modconfig_DATA)
 
 include $(top_srcdir)/Makefile.am.common
diff --git a/doc/autodocs/Makefile.am b/doc/autodocs/Makefile.am
index 2a6f678..9de7be0 100644
--- a/doc/autodocs/Makefile.am
+++ b/doc/autodocs/Makefile.am
@@ -1,3 +1,5 @@
 # Makefile.am for YCP module .../doc/autodocs
 
-include $(top_srcdir)/autodocs-ycp.ami
+# TODO FIXME: temporarily disabled
+
+#include $(top_srcdir)/autodocs-ycp.ami
diff --git a/src/product_start.ycp b/src/product_start.ycp
deleted file mode 100644
index 671b84b..0000000
--- a/src/product_start.ycp
+++ /dev/null
@@ -1,83 +0,0 @@
-/* ------------------------------------------------------------------------------
- * Copyright (c) 2012 Novell, Inc. All Rights Reserved.
- *
- *
- * This program is free software; you can redistribute it and/or modify it under
- * the terms of version 2 of the GNU General Public License as published by the
- * Free Software Foundation.
- *
- * This program is distributed in the hope that it will be useful, but WITHOUT
- * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
- * FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
- *
- * You should have received a copy of the GNU General Public License along with
- * this program; if not, contact Novell, Inc.
- *
- * To contact Novell about this file by physical or electronic mail, you may find
- * current contact information at www.novell.com.
- * ------------------------------------------------------------------------------
- */
-
-/**
- * Maintainer:             Jiri Srain <jsrain@suse.cz>
- *
- * $Id$
- */
-{
-    textdomain "firstboot";
-
-    import "Mode";
-    import "Stage";
-    import "ProductControl";
-    import "Wizard";
-    import "Report";
-    import "Directory";
-
-    string control = "";
-    string start_dir = "";
-    if (size (WFM::Args ()) > 0 && (is (WFM::Args(0), string ) && is(WFM::Args(1), string)))
-    {
-        control  = (string)WFM::Args(0);
-        start_dir = (string)WFM::Args(1);
-        if (SCR::Read(.target.size, control )<= 0 )
-        {
-            y2error("file  %1 empty or not found", control);
-            return `abort;
-        }
-    }
-
-
-    Directory::custom_workflow_dir = start_dir;
-
-
-    Mode::SetMode("installation");
-    y2milestone("Current mode: %1", Mode::mode());
-
-    ProductControl::custom_control_file = control;
-    if (!ProductControl::Init())
-    {
-        y2error("control file %1 not found", ProductControl::custom_control_file );
-        return `abort;
-    }
-
-    Wizard::OpenNextBackStepsDialog();
-
-    string wizard_mode = Mode::test () ? "installation" : Mode::mode ();
-
-    list<map> stage_mode = [$["stage": "normal",  "mode": wizard_mode ]];
-    ProductControl::AddWizardSteps(stage_mode);
-
-    // Do log Report messages by default (#180862)
-    Report::LogMessages(true);
-    Report::LogErrors(true);
-    Report::LogWarnings(true);
-
-
-    symbol ret = ProductControl::Run();
-
-    Pkg::SourceFinishAll();
-    Pkg::TargetFinish();
-
-    UI::CloseDialog();
-    return ret;
-}
diff --git a/yast2-firstboot.spec.in b/yast2-firstboot.spec.in
index b0a1cd5..8a44fb3 100644
--- a/yast2-firstboot.spec.in
+++ b/yast2-firstboot.spec.in
@@ -20,6 +20,8 @@ Requires:	yast2-installation >= 2.19.0
 
 BuildArchitectures:	noarch
 
+Requires:       yast2-ruby-bindings >= 1.0.0
+
 Summary:	YaST2 - Initial System Configuration
 PreReq:         %fillup_prereq
 
@@ -53,10 +55,10 @@ mkdir -p $RPM_BUILD_ROOT/usr/share/firstboot/scripts
 %dir @ystartupdir@/startup/Firstboot-Stage
 @ystartupdir@/startup/Firstboot-Stage/*
 @ystartupdir@/startup/YaST2.Firstboot
-@clientdir@/firstboot_*.ycp
-@clientdir@/firstboot.ycp
+@clientdir@/firstboot_*.rb
+@clientdir@/firstboot.rb
 %dir @yncludedir@/firstboot
-@yncludedir@/firstboot/*.ycp
+@yncludedir@/firstboot/*.rb
 @moduledir@/Firstboot.*
 @scrconfdir@/*.scr
 /var/adm/fillup-templates/sysconfig.firstboot
